contrib/signet/miner

```diff
@@ -250,6 +254,42 @@ class Generate:
         det_rand = int(last_hash[-16:-8], 16)
         return self.multi_low <= (det_rand % self.multi_period) < self.multi_high
 
+    def next_block_time(self, now, bestheader, is_first_block):
+        if self.set_block_time is not None:
+            logging.debug("Setting start time to %d", self.set_block_time)
+            self.mine_time = self.set_block_time
+            self.action_time = now
+            self.is_mine = True
+        elif bestheader["height"] == 0:
+            time_delta = self.INTERVAL * 100 # plenty of time to mine 100 blocks
```

Comment:
In 85c5c0bea9d45e93a9fb20988457480798d68637 "signet/miner: add Generate.next_block_time function"

Why change the time delta to use `self.INTERVAL` instead of using `next_block_delta` as previously?

Reply:
`next_block_delta` can be somewhat small if you have a high difficulty target (ie 2.5m instead of 10m) and if `--poisson` is specified will be smaller again (the deterministic random calculation used against the genesis block reduces the target by ~73%). The `--poisson` result alone is already annoying -- it will generally only take 25 blocks before it's caught up to real time; but even without that if you switch from mining with a high nbits target to a minimal nbits target, then even without `--poisson` you'll also fail to mine 100 blocks before catching up to real time.